addon/helpers/changeset.js

```diff
@@ -2,14 +2,23 @@ import Ember from 'ember';
 import Changeset from 'ember-changeset';
 import lookupValidator from 'ember-changeset-validations';
 import isObject from 'ember-changeset/utils/is-object';
+import isPromise from 'ember-changeset/utils/is-promise';
 
 const { Helper: { helper } } = Ember;
 
 export function changeset([model, validationMap]) {
   if (isObject(validationMap)) {
+    if (isPromise(model)) {
+      model.then((resolved) => new Changeset(resolved, lookupValidator(validationMap), validationMap));
```

Comment:
Sorry for getting back to this so late, but you should try https://github.com/fivetanley/ember-promise-helpers in conjunction